app/src/main/java/br/org/cesar/madmovies/movies/viewmodel/MovieDetailsViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package br.org.cesar.madmovies.movies.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import br.org.cesar.madmovies.movies.data.remote.RemoteRepository
+import br.org.cesar.madmovies.movies.domain.model.Movie
+import br.org.cesar.madmovies.movies.domain.usecase.GetMovieDetail
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class MovieDetailsViewModel : ViewModel() {
+
+    var movieId: Int = -1
+    private var _movieFlow: MutableStateFlow<Movie> = MutableStateFlow(Movie())
+    var movieFlow: StateFlow<Movie> =
+        _movieFlow
+            .onSubscription {
+                if (_movieFlow.value.id != movieId) {
+                    getMovieDetail(movieId)
+                }
+            }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(), _movieFlow.value)
+
+    fun getMovieDetail(movieId: Int) {
+        viewModelScope.launch {
```

Comment:
_movieFlow is a MutableStateFlow, meaning you can just set .value synchronously (no coroutine needed) instead of emitting

Reply:
yeah but the GetMovieList use case needs to be run in a coroutine. So I can't just update the flow outside the coroutine, if the value is acquired inside the coroutine.

Reply:
Ok